packages/components/app/styles/components/dropdown.scss

```diff
@@ -156,171 +162,194 @@ $size-props: (
-      border-color: transparent;
-      padding: 0 4px 0 10px;
-      position: relative;
-      font-size: 14px;
-      line-height: 20px;
+      outline-style: solid; // used to avoid double outline+focus-ring in Safari (see https://github.com/hashicorp/design-system-components/issues/161#issuecomment-1031548656)
+      outline-color: transparent;
+      padding: 8px 10px 8px 16px;
+      text-decoration: none;
 
+      // this is used for the left "hover" indicator
       &::before {
         border-radius: 1px;
-        bottom: 4px;
+        bottom: 6px;
         content: '';
         left: 4px;
         position: absolute;
-        top: 4px;
+        top: 6px;
         width: 2px;
+        z-index: -1;
       }
 
-      a {
-        @include hds-focus-ring-with-pseudo-element($top: 0, $right:0, $bottom: 0, $left: 0, $radius: 5px); //TODO @didoo need critical variant
-        border-radius: $hds-dropdown-list-item-border-radius;
-        display: grid;
-        font-size: 15px;
-        gap: 0px 6px;
-        grid-auto-flow: row;
-        grid-template-columns: auto 1fr;
-        grid-template-rows: min-content;
-        line-height: 20px;
-        max-width: 100%;
-        padding: 10px;
-        text-decoration: none;
-        width: 100%;
+      // Notice: this is used for the active/focus states which have very specific positions
+      // and also has a background color, so we can't use the existing focus-ring mixins
+      &::after {
+        border-radius: 5px;
+        bottom: 0px;
+        content: '';
+        left: 10px;
+        position: absolute;
+        right: 4px;
+        z-index: -1;
+        top: 0;
       }
 
-      &.hds-dropdown-list-item__link--action {
-        &:hover {
-          &::before {
+      // Notice: to avoid too much duplication we define two local CSS variables
+      // and define their values in the color variants below
+
+      // default focus for browsers that still rely on ":focus"
+      &:focus,
+      &.is-focus {
+        &::after {
+          background-color: var(--current-background-color);
+          box-shadow: var(--current-focus-ring-box-shadow);
+          left: 4px;
+        }
+      }
+      // undo the previous declaration for browsers that support ":focus-visible" but wouldn't normally show default focus styles
+      &:focus:not(:focus-visible) {
+        &::after {
+          box-shadow: none;
+        }
+      }
+      // set focus for browsers that support ":focus-visible"
+      &:focus-visible {
+        &::after {
+          background-color: var(--current-background-color);
+          box-shadow: var(--current-focus-ring-box-shadow);
+          left: 4px;
+        }
+      }
+      // remove the focus ring on "active + focused" state (by design) // TODO Mel fix this
+      &:focus:active,
+      &.is-focus.is-active {
+        &::after {
+          box-shadow: none;
+        }
+      }
+    }
+  
+    &.hds-dropdown-list-item--link--color-action {
```

Comment:
[issue] I think one of the conflicts fixes after a merge have removed a big chunk of code, now all the styling is gone. you can see how it was done before in this commit: https://github.com/hashicorp/design-system/pull/68/commits/cf72582efb2a5bf4d3421aa4d148cfcbbdb406d2